ROADMAP.md

```diff
@@ -1,25 +1,389 @@
+
+* Provide attestations of trusted actions, such as verifying identity tokens, to minimize needed trust in service operators
+* Use modern cryptography (zero-knowledge proofs, verifiable random functions) to remove PII where possible
+
+### **Ensure that verification is as seamless as signing**
+
+* Integrate with existing verification policy languages (e.g. Rego, CUE)
+* Provide a reference implementation for how to verify signatures across various points in the software development lifecycle
+
+### **Offer intuitive and ergonomic tooling for engineers and cloud operations**
+
+* [Cosign CLI](https://github.com/sigstore/cosign) is the official Sigstore CLI multi-tool for signing and verification in a variety of environments and use cases.
+* [sigstore-policy-controller](https://github.com/sigstore/policy-controller) is the center of Sigstore-oriented verification flows in Kubernetes environments.
+
+# Roadmap
+
+## Client SDKs
+
+### Short Term (< 6 mo)
+
+* Releases:
+    * sigstore-java v1 release
+    * sigstore-go beta release
+    * sigstore-rust v1 release
+    * sigstore-python v2 release
+* FFI Design
+* Support for bundle format across all Sigstore clients
+* Simplifying verification policy
+    * Improve Cosign UX for specifying OIDC identities
+    * Supporting CI claims
+* Complete architecture documentation for clients
+* Client maturity criteria
+* All clients are integrated with the conformance test suite. All tests may not succeed, but tests are run periodically
+
+### Medium Term (6-12 mo)
+
+* FFI Implementation
+* Port at least one language to make use of FFI implementation
+* Releases:
+    * sigstore-go v1 release
+    * Gitsign v1 release
+    * policy-controller v1 release
+* Supporting BYO PKI and private deployments in Sigstore clients
+* Improved testing
+    * End-to-end testing
+    * Conformance tests are passing for all clients
+    * Fuzz tests
+* Improving verification policies
+    * Rego/CUE integration
+    * [Hammurabi from web PKI](https://www.andrew.cmu.edu/user/bparno/papers/hammurabi.pdf)
+* Reduce size of Cosign binary, fewer dependencies
+* Privacy-preserving telemetry (https://divviup.org/)
+
+### Long Term (12+ mo)
+
+* Plumbing and porcelain model for Cosign, to improve Cosign's user experience
+* Policy: sophisticated verification policies, or deeper integration with a policy evaluation service
+* Further integration with TUF through delegations or trusted verification material distribution, and in-toto for signed provenance
+* Adding additional conformance tests to verify consistent behavior across clients, including negative tests
+* Bundle format includes attestation support that interops with SLSA
+* Reference implementation of how to use signature verification across various points in the software development lifecycle, leveraging policy-controller as one example
+
+## Signature Transparency Log (Rekor)
+
+### Short Term (< 6 mo)
+
+* Add 2 independent log monitors
```

Comment:
Should we call out what we want to use to monitor the logs? I know we have a monitoring project here in the Sigstore org that I believe powers the Purdue monitor. Should there be a roadmap for that kind of functionality?

Reply:
Good suggestion, I’ll add a section on that. I have a roadmap for what’s needed to enable witnessing and identity monitoring.

Reply:
Added roadmap, PTAL!